src/Advection/vector_invariant_advection.jl

```diff
@@ -38,142 +32,173 @@ Keyword arguments
-                     argument has no effect). In case `divergence_scheme` is an `AbstractUpwindBiasedAdvectionScheme`, 
-                     `vertical_scheme` describes a flux form reconstruction of vertical momentum advection, and any 
-                     advection scheme can be used - `Centered`, `UpwindBiased` and `WENO` (defaults to `EnergyConservingScheme`)
+- `vertical_scheme`: Scheme used for vertical advection of horizontal momentum and upwinding of divergence and kinetic energy gradient. Defaults to `EnergyConservingScheme`)
+- `upwinding`: Treatment of upwinding in case of Upwinding reconstruction of divergence and kinetic energy gradient. Choices are between
+                         `CrossAndSelfUpwinding`, `OnlySelfUpwinding` and `VelocityUpwinding` (defaults to `OnlySelfUpwinding`)
+- `multi_dimensional_stencil` : if true, use a horizontal two dimensional stencil for the reconstruction of vorticity, divergence and kinetic energy gradient.
+                                The tangential direction is _always_ treated with a 5th order centered WENO reconstruction
 
 Examples
 ========
 ```jldoctest
 julia> using Oceananigans
 
 julia> VectorInvariant()
-Vector Invariant reconstruction, maximum order 1 
+Vector Invariant, Dimension-by-dimension reconstruction 
  Vorticity flux scheme: 
-    └── EnstrophyConservingScheme{Float64} 
- Divergence flux scheme: 
-    └── Nothing 
- Vertical advection scheme: 
-    └── EnergyConservingScheme{Float64}
+ └── EnstrophyConservingScheme{Float64} 
+ Vertical advection / Divergence flux scheme: 
+ └── EnergyConservingScheme{Float64}
 
 ```
 ```jldoctest
 julia> using Oceananigans
 
-julia> VectorInvariant(vorticity_scheme = WENO(), divergence_scheme = WENO(), vertical_scheme = WENO(order = 3))
-Vector Invariant reconstruction, maximum order 5 
+julia> VectorInvariant(vorticity_scheme = WENO(), vertical_scheme = WENO(order = 3))
+Vector Invariant, Dimension-by-dimension reconstruction 
  Vorticity flux scheme: 
-    └── WENO reconstruction order 5 with smoothness stencil Oceananigans.Advection.VelocityStencil()
- Divergence flux scheme: 
-    └── WENO reconstruction order 5 with smoothness stencil Oceananigans.Advection.DefaultStencil()
- Vertical advection scheme: 
-    └── WENO reconstruction order 3
+ ├── WENO reconstruction order 5 
+ └── smoothness ζ: Oceananigans.Advection.VelocityStencil()
+ Vertical advection / Divergence flux scheme: 
+ ├── WENO reconstruction order 3
+ └── upwinding treatment: OnlySelfUpwinding 
+ KE gradient and Divergence flux cross terms reconstruction: 
+ └── Centered reconstruction order 2
+ Smoothness measures: 
+ ├── smoothness δU: FunctionStencil f = divergence_smoothness
+ ├── smoothness δV: FunctionStencil f = divergence_smoothness
+ ├── smoothness δu²: FunctionStencil f = u_smoothness
+ └── smoothness δv²: FunctionStencil f = v_smoothness
```

Comment:
```suggestion
 ├── δU: FunctionStencil f = divergence_smoothness
 ├── δV: FunctionStencil f = divergence_smoothness
 ├── δu²: FunctionStencil f = u_smoothness
 └── δv²: FunctionStencil f = v_smoothness
```

there is a suggestion to change the show further down....  :)

Reply:
as above

Reply:
but there is a title over these saying "smoothness", right?

Reply:
OK, if you insist to keep the "smoothness" in every line at least put it after the variable, e.g., 
```Julia
 ├── δU smoothness: FunctionStencil f = divergence_smoothness
```
?